tests/operators/test_azure_container_instance.py

```diff
@@ -0,0 +1,110 @@
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+from airflow.utils.context import Context
+from pendulum import datetime
+
+from cosmos.operators.azure_container_instance import (
+    DbtAzureContainerInstanceBaseOperator,
+    DbtLSAzureContainerInstanceOperator,
+    DbtRunAzureContainerInstanceOperator,
+    DbtTestAzureContainerInstanceOperator,
+    DbtSeedAzureContainerInstanceOperator,
+)
+
+
+def test_dbt_azure_container_instance_operator_add_global_flags() -> None:
+    dbt_base_operator = DbtAzureContainerInstanceBaseOperator(
+        ci_conn_id="my_airflow_connection",
+        task_id="my-task",
+        image="my_image",
+        region="Mordor",
+        name="my-aci",
+        resource_group="my-rg",
+        project_dir="my/dir",
+        vars={
+            "start_time": "{{ data_interval_start.strftime('%Y%m%d%H%M%S') }}",
+            "end_time": "{{ data_interval_end.strftime('%Y%m%d%H%M%S') }}",
+        },
+        no_version_check=True,
+    )
+    assert dbt_base_operator.add_global_flags() == [
+        "--vars",
+        "end_time: '{{ data_interval_end.strftime(''%Y%m%d%H%M%S'') }}'\n"
+        "start_time: '{{ data_interval_start.strftime(''%Y%m%d%H%M%S'') }}'\n",
+        "--no-version-check",
+    ]
+
+
+@patch("cosmos.operators.base.context_to_airflow_vars")
+def test_dbt_azure_container_instance_operator_get_env(p_context_to_airflow_vars: MagicMock) -> None:
+    """
+    If an end user passes in a
+    """
+    dbt_base_operator = DbtAzureContainerInstanceBaseOperator(
+        ci_conn_id="my_airflow_connection",
+        task_id="my-task",
+        image="my_image",
+        region="Mordor",
+        name="my-aci",
+        resource_group="my-rg",
+        project_dir="my/dir",
+    )
+    dbt_base_operator.env = {
+        "start_date": "20220101",
+        "end_date": "20220102",
+        "some_path": Path(__file__),
+        "retries": 3,
+        ("tuple", "key"): "some_value",
+    }
+    p_context_to_airflow_vars.return_value = {"START_DATE": "2023-02-15 12:30:00"}
+    env = dbt_base_operator.get_env(
+        Context(execution_date=datetime(2023, 2, 15, 12, 30)),
+    )
+    expected_env = {
+        "start_date": "20220101",
+        "end_date": "20220102",
+        "some_path": Path(__file__),
+        "START_DATE": "2023-02-15 12:30:00",
+    }
+    assert env == expected_env
```

Comment:
If the env vars can be set for the ACI, then it would be worth having a check that the environment variables are set in the `build_command` method.

Reply:
Yeah good one. I added the `environment_variables` to the `build_command` method, and added a separate test to validate the behaviour. I think this is what you meant?